frontend/src/metabase-lib/lib/metadata/Database.ts

```diff
@@ -18,12 +17,15 @@ import Metadata from "./Metadata";
  */
 
 export default class Database extends Base {
-  id: number;
-  name: string;
-  description: string;
-  tables: Table[];
-  schemas: Schema[];
-  metadata: Metadata;
+  id!: number;
```

Comment:
I'm on the fence about using `!` to guarantee a property's existence. The cause is that we depend on `Base` to assign properties, so TypeScript can't figure out that these properties actually exist on the instance, not without an explicit `constructor ({ id }) { this.id = id; }` and so on.

If people feel strongly that we should not do this, I'm happy to change it.